subworkflows/local/bgc.nf

```diff
@@ -149,6 +152,13 @@ workflow BGC {
 
     // COMBGC
     COMBGC ( ch_bgcresults_for_combgc )
+
+    ch_bgcresults_for_combgc.ifEmpty(
+        fastas.filter {
+            meta, fna ->
+                log.warn("[nf-core/funcscan] BGC workflow: No hits found by BGC tools; comBGC summary tool will not be run for sample ${meta.id}.")
+        }
```

Comment:
Again, I don't like putting operators within operators, I can't actually understand what this is doing? There is no actual filtering taking place here, why is the purpose of the filter?

Also why is this filtering (if it is actually filtering)  of the channel on input to COMBGC placed _after_ the actual invocation of COMBGC itself?

Reply:
The filtering is only to get access to the current `meta` for the warning string. Is there an easier way?

Reply:
I would use `map`

Reply:
Finally fixed in [18afe1f](https://github.com/nf-core/funcscan/pull/386/commits/18afe1fcfd040a34e541cf5d9a16f5a740db4b53).